ledger/internal/eval.go

```diff
@@ -1107,6 +1108,10 @@ func (eval *BlockEvaluator) transaction(txn transactions.SignedTxn, evalParams *
 	// Apply the transaction, updating the cow balances
 	applyData, err := eval.applyTransaction(txn.Txn, cow, evalParams, gi, cow.Counter())
 	if err != nil {
+		if eval.Tracer != nil {
+			// If there is a tracer, save the ApplyData so that it's viewable by the tracer
+			txib.ApplyData = applyData
```

Comment:
Without this change it's impossible for the tracer to see the partially built up ApplyData on failure. Hopefully we can agree this is harmless

Reply:
is there a thought to only performing this if the error is of a certain type and not for others

Reply:
No, I think this is useful for every error type I'm aware of

Reply:
my thought here was that we could avoid going down additional codepaths if the error is some kind of resource starvation but it's ok.